ipynb/core-balance-classifier.ipynb

```diff
@@ -0,0 +1,1078 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here we will create balance classifiers to identify differences due to plant diet and country.\n",
+    "\n",
+    "Make sure that the appropriate files are downloaded.  They can be obtained from the ftp link using the following commands\n",
+    "\n",
+    "```bash\n",
+    "wget ftp://ftp.microbio.me/AmericanGut/manuscript-package/1250/deblur_125nt_no_blooms.biom\n",
+    "\n",
+    "wget ftp://ftp.microbio.me/AmericanGut/manuscript-package/1250/ag_map_with_alpha.txt\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "from biom import load_table\n",
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np\n",
+    "from skbio import TreeNode\n",
+    "import seaborn as sns\n",
+    "from patsy import dmatrix\n",
+    "\n",
+    "%matplotlib inline\n",
+    "sns.set_style('ticks')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/Users/mortonjt/miniconda3/envs/bio/lib/python3.5/site-packages/IPython/core/interactiveshell.py:2698: DtypeWarning: Columns (0,66) have mixed types. Specify dtype option on import or set low_memory=False.\n",
+      "  interactivity=interactivity, compiler=compiler, result=result)\n"
+     ]
+    }
+   ],
+   "source": [
+    "from gneiss.util import match_tips\n",
+    "\n",
+    "table = load_table('deblur_125nt_no_blooms.biom')\n",
+    "\n",
+    "obs_ids = table.ids(axis='observation')\n",
+    "taxa = [table.metadata(id=i, axis='observation')['taxonomy'] for i in obs_ids]\n",
+    "taxa = pd.DataFrame(taxa, index=obs_ids, \n",
+    "                    columns=['kingdom', 'phylum', 'class', 'order', \n",
+    "                             'family', 'genus', 'species'])\n",
+    "\n",
+    "# we will propagate the last annotation\n",
+    "def taxa_fix(x):\n",
+    "    suffixes = ['k__', 'p__', 'c__', 'o__', 'f__', 'g__', 's__']\n",
+    "    for i, xi in enumerate(x):\n",
+    "        if xi in suffixes:\n",
+    "            x[i] = x[i-1]\n",
```

Comment:
This impacts the generated tables of differential sOTUs which have incorrect taxon strings, and which need to be corrected for the submission